Formula/john-jumbo.rb

```diff
@@ -53,12 +53,25 @@ class JohnJumbo < Formula
     sha256 "65a4aacc22f82004e102607c03149395e81c7b6104715e5b90b4bbc016e5e0f7"
   end
 
+  # Upstream M1/ARM64 Support.
+  # Combined diff of the following four commits, minus the doc changes
+  # that block this formula from using these commits otherwise.
+  # https://github.com/openwall/john/commit/d6c87924b85323b82994ce01724d6e458223fd36
+  # https://github.com/openwall/john/commit/d531f97180a6e5ae52e21db177727a17a76bd2b4
+  # https://github.com/openwall/john/commit/c9825e688d1fb9fdd8942ceb0a6b4457b0f9f9b4
+  # https://github.com/openwall/john/commit/716279addd5a0870620fac8a6e944916b2228cc2
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/50a00afbf4549fbc0ffd3855c884f7d045cf4f93/john-jumbo/john_jumbo_m1.diff"
+    sha256 "6658f02056fd6d54231d3fdbf84135b32d47c09345fc07c6f861a1feebd00902"
+  end
+
   def install
     ENV.append "CFLAGS", "-DJOHN_SYSTEMWIDE=1"
     ENV.append "CFLAGS", "-DJOHN_SYSTEMWIDE_EXEC='\"#{share}/john\"'"
     ENV.append "CFLAGS", "-DJOHN_SYSTEMWIDE_HOME='\"#{share}/john\"'"
 
-    ENV.append "CFLAGS", "-mno-sse4.1" unless MacOS.version.requires_sse4?
+    # Apple's M1 chip has no support for SSE 4.1.
+    ENV.append "CFLAGS", "-mno-sse4.1" if Hardware::CPU.intel? && !MacOS.version.requires_sse4?
```

Comment:
wouldn't it need to be added on M1 too? Like:
```suggestion
    # Apple's M1 chip has no support for SSE 4.1.
    ENV.append "CFLAGS", "-mno-sse4.1" unless MacOS.version.requires_sse4? || Hardware::CPU.intel?
```

Reply:
Nah, SSE doesn't exist on ARM so it's impossible for the compiler to even try use it. `-mno-sse4.1` possibly even errors on ARM64.

Reply:
so... it's not `-m no-sse4.1` then? I was expecting this to disable the feature

Reply:
SSE flags do nothing on non-x86_64 platforms.

I thought they might error but it seems they warn instead in Clang:

```
argument unused during compilation: '-mno-sse4.1' [-Wunused-command-line-argument]
```

On GCC, it is a hard error on non-x86_64 platforms:

```
aarch64-unknown-linux-gnu-g++: error: unrecognized command-line option '-mno-sse4.1'
```

Reply:
Homebrew itself stops you even trying on ARM/Apple Silicon, which is reasonable but not immediately obvious what's happened to the end user:
```
Error: An exception occurred within a child process:
  RuntimeError: Unexpected architecture: arm64. This only works with Intel architecture.
```

Reply:
Ah, yes. Too busy looking at the flag without realising `MacOS.version.requires_sse4?` errors.